NodeJS/src/AddressBookClient.ts

```diff
@@ -0,0 +1,46 @@
+import { SkillPriority } from './models'
+import { RestClient } from './RestClient'
+import { sleep } from './Utils'
+import { AddressBookResponse } from './models/AddressBookResponse'
+import { AddressBookSearchRequest } from './models/AddressBookSearchRequest'
+import { AddressBookSearchResponse } from './models/AddressBookSearchResponse'
+
+export class AddressBookClient {
+  private restClient: RestClient
+  private subAccountId: string
+
+  constructor(subAccountId: string, restClient: RestClient) {
+    this.restClient = restClient
+    this.subAccountId = subAccountId
+  }
+
+  /**
+   * @returns AddressBookResponse with all active contacts for a Sub-account
+   */
+  public async getAddressBook(): Promise<AddressBookResponse> {
+    return await this.restClient.getAddressBook()
+  }
+
+  /**
+   * @param AddressBookSearchRequest (optional) search by type, query, orderBy, orderDirection, page, pageSize
+   * @returns AddressBookSearchResponse with active contacts for a Sub-account according to search params
+   */
+  public async searchContacts(
+    request?: AddressBookSearchRequest
+  ): Promise<AddressBookSearchResponse> {
+    return await this.restClient.searchContacts(request)
+  }
+}
+
+async function createInstance(restClient: RestClient) {
+  const subAccountId = await restClient.getSubAccountId()
```

Comment:
not `getSubAccountAppId`?

Reply:
correct, it's clientId, which is useless for that controller. will refactor.

Reply:
removed